mantidproject/mantid
# Fix bug causing empty runs to be subtracted twice in Polaris.focus

re: #27699

**Description of work.**
This PR fixes a bug in Polaris.focus that resulted in empty runs to be subtracted twice if the user specifies a sample_empty, the function would subtract first the empty runs specified in the config maps, then the runs specified in sample_empty.

**Report to:** Helen Playford

**To test:**

1. contact me on slack for the relivant files to run the script below.
1. open workbench and run the script.
2. rename the `97947-ResultD` group and its contents with the suffix '_default_empty'.
3. comment out the two call of `Polaris.focus` in the script and uncomment out the third.
4. run the script again creating a new `97947-ResultD` group.
5. run the Minus algorithm subtracting the `97948-97951-ResultD` group from the `97947-ResultD_default_empty` group
6. Compare the `97947-ResultD` group to the output from the minus algorithm. they should be identical.

```
# The following line helps with future compatibility with Python 3
# print must now be used as a function, e.g print('Hello','World')
from __future__ import (absolute_import, division, print_function, unicode_literals)

# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *

import numpy as np

from isis_powder import Polaris, SampleDetails

config_file_path = r"C:/Users/wey38795/Documents/TS_dev/polaris_config_example.yaml"

###

polaris = Polaris(config_file=config_file_path, user_name="Test", mode="PDF")  

sample_details = SampleDetails(height=1.0, radius=0.2985, center=[0.0, 0.0, 0.0], shape='cylinder')
sample_details.set_material(chemical_formula='Si',number_density=0.05)
polaris.set_sample_details(sample=sample_details)

polaris.focus(do_absorb_corrections=False,do_van_normalisation=True, input_mode="Individual",run_number="97947")
polaris.focus(do_absorb_corrections=False,do_van_normalisation=True, input_mode="Summed",run_number="97948-97951")
#polaris.focus(do_absorb_corrections=False,do_van_normalisation=True, input_mode="Individual",run_number="97947", 
#    sample_empty="97948-97951",sample_empty_scale=1.0)
```

Fixes #27699 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
